### PR TITLE
Modifiers on new selection UI

### PR DESCRIPTION
**Pull request by talonateox**
The implementation is very crude BUT functional. If anything is not to your liking just give me a review

CHANGES:
- Modifiers UI
- Speed Modifier
- Nofail modifier
- Reset modifiers
- Seek slider
- Fix pages bug where when there are no search results, pages would be 0, then when results come back it will STAY on 0, which now it automatically goes to the first valid page

### CODE REVIEW

**Comment by AllusionistZ** [comment]
Guys. This is awesome.

File: scripts/menu/maplist/MapDetails.gd [hunk 1]
```diff
@@ -9,7 +9,7 @@ signal map_selected
 
 var difficulties = []
 
-var selected_mapset
+@export var selected_mapset: Mapset = null
```
Comment (kermeow) [inline scripts/menu/maplist/MapDetails.gd]: what is this for

File: scripts/menu/maplist/Modifiers.gd [hunk 2]
```diff
@@ -2,6 +2,41 @@ extends Control
...
+
+func _set_speed(value):
+	Rhythia.selected_mods.speed_custom = value
 
 func _on_reset_modifiers():
-	pass
+	Rhythia.selected_mods.start_from = 0.0
+	Rhythia.selected_mods.speed_custom = 1.0
+	Rhythia.selected_mods.no_fail = false
+	_ready()
```
Comment (kermeow) [inline scripts/menu/maplist/Modifiers.gd]: this makes connect() run again

File: scenes/Init.tscn [hunk 3]
```diff
@@ -1,7 +1,7 @@
 [gd_scene load_steps=7 format=3 uid="uid://eobt2fotkbfh"]
 
 [ext_resource type="Script" path="res://scripts/menu/loading/Init.gd" id="1"]
-[ext_resource type="Theme" uid="uid://diarojdcbhxg8" path="res://default_theme.tres" id="1_3bv2c"]
+[ext_resource type="Theme" uid="uid://tin0gsmmldph" path="res://themes/modifier_theme.tres" id="1_mls4r"]
```
Comment (kermeow) [inline scenes/Init.tscn]: why is the theme changed in this scene?

File: scripts/menu/maplist/MapDetails.gd [hunk 4]
```diff
@@ -70,6 +70,10 @@ func _update_details():
 		selected_mapset.id,
 		"local" if selected_mapset.local else selected_mapset.online_id
 	]
+	
+	Rhythia.selected_mods.start_from = min(Rhythia.selected_mods.start_from, selected_mapset.length as float)
+	$Sections/Extra/Modifiers.update_seek_slider()
+	$Sections/Extra/Modifiers.update_time_label()
```
Comment (kermeow) [inline scripts/menu/maplist/MapDetails.gd]: this should be done in the modifiers script

File: scenes/Menu.tscn [hunk 5]
```diff
@@ -1,7 +1,7 @@
-[gd_scene load_steps=36 format=3 uid="uid://d1x4owtlent1e"]
+[gd_scene load_steps=37 format=3 uid="uid://d1x4owtlent1e"]
 
 [ext_resource type="PackedScene" uid="uid://de1sj0rkm2iu7" path="res://prefabs/menu/VersionNumberR.tscn" id="1"]
-[ext_resource type="Theme" uid="uid://diarojdcbhxg8" path="res://default_theme.tres" id="1_kkcab"]
+[ext_resource type="Theme" uid="uid://tin0gsmmldph" path="res://themes/modifier_theme.tres" id="1_j808x"]
```
Comment (kermeow) [inline scenes/Menu.tscn]: same with the init scene? this should not be replacing the main theme

**Review by kermeow — CHANGES_REQUESTED** [review]
Please see comments.

**Comment by talonateox** [comment]
Done, review the commits

File: scenes/Init.tscn [hunk 6]
```diff
@@ -55,6 +54,7 @@ text = "Continue"
 anchors_preset = 0
 anchor_right = 1.0
 anchor_bottom = 1.0
+theme = ExtResource("2_p8tgo")
```
Comment (kermeow) [inline scenes/Init.tscn]: This is not the root of the scene what

File: scripts/menu/maplist/Modifiers.gd
Comment (kermeow) [inline scripts/menu/maplist/Modifiers.gd:1]: this still calls _ready() again please dont do this

Comment (talonateox) [reply scripts/menu/maplist/Modifiers.gd:1]: where??? you can check the commit for it, i literally removed the call for _ready()

**Review by kermeow — CHANGES_REQUESTED** [review]
see comments again, some things are ok others are ???

**Review by kermeow — APPROVED** [review]
good enough but i might need to make some code guidelines in the future :3

**Review by kiwirew — APPROVED** [review]

**Pull request merged**